### PR TITLE
Update doc to reflect new repo structure (dev as default).

### PR DESCRIPTION
**Pull request by janiversen**
In order to have "dev" as default branch the README.rst and INSTALL needed to be updated. 

While doing it I have added/corrected text to reflect our current status (including updating TODO).

With this PR merged, default branch should be shifted to "dev" (this can only be done by people with "maintainer" role == @dhoomakethu )

Remark: English is not my primary language, I am danish a real viking living in souther Spain, so please feel free to suggest different wordings.

### CODE REVIEW

**Comment by janiversen** [comment]
@dhoomakethu Can you please review this, it matches how most projects do releases (from my relative broad experience), but of course it is only a suggestion. An alternative would be to keep master for series 2.x and make "master3) for series 3.x, but I do not really see the need.

@swamper123 your opinion is also wanted.

**Comment by dhoomakethu** [comment]
Thanks @janiversen appreciate you taking time on a sunday. I am on mobile and will update the branch and merge this jira tomorrow. 
P.S English is not a first language for me neither.

File: README.rst [hunk 1]
```diff
@@ -13,21 +11,34 @@ PyModbus - A Python Modbus Stack
...
-=======
-(including python 3+)
->>>>>>> 3.0.0
+Pymodbus is a full Modbus protocol implementation using a synchronous or asynchronous core. The preferred mode for asynchronous communication is asyncio, however for the moment twisted and tornado are also supported (due to be removed or converted to a plugin in a later version).
+
+Supported modbus communication modes:
+- tcp
+- rtuovertcp
+- udp
+- serial
```
Comment (dhoomakethu) [inline README.rst:35]: Also support `tls`

File: README.rst [hunk 2]
```diff
@@ -39,24 +50,25 @@ Client Features
...
   * Pymodbus REPL for quick tests
+  * Customable framer to allow for custom implementations
 
 ~~~~~~~~~~~~~~~~~~~~
 Server Features
 ~~~~~~~~~~~~~~~~~~~~
 
   * Can function as a fully implemented modbus server
-  * TCP, UDP, Serial ASCII, Serial RTU, and Serial Binary
+  * TCP, RTUOVERTCP, UDP, Serial ASCII, Serial RTU, and Serial Binary
```
Comment (dhoomakethu) [inline README.rst:64]: Can we change `RTUOVERTCP` to `RTU-OVER-TCP` for the sake of readability

File: README.rst [hunk 3]
```diff
@@ -157,34 +175,57 @@ Otherwise you can pull the trunk source and install from there::
 
     git clone git://github.com/bashwork/pymodbus.git
     cd pymodbus
+    
+To get latest release::
```
Comment (dhoomakethu) [inline README.rst:179]: Let's be explicit and say for now the latest release means `2.x` pymobuds with python 2 support.

File: doc/TODO [hunk 4]
```diff
@@ -8,46 +8,34 @@ General
...
+- get coverage > 98%
+- clean raise to be at frame level
 
 ---------------------------------------------------------------------------
 Protocols
 ---------------------------------------------------------------------------
 
 - Serial RTU -> just use sleep wait
-- Test serial against devices (and virtual tty)
+- Write unitfest for serial with virtual tty
```
Comment (dhoomakethu) [inline doc/TODO:27]: `unittest` I suppose ?

File: README.rst [hunk 5]
```diff
@@ -2,8 +2,6 @@
 PyModbus - A Python Modbus Stack
 ================================
 
-.. image:: https://travis-ci.org/riptideio/pymodbus.svg?branch=master
```
Comment (dhoomakethu) [inline README.rst:5]: We should add github badge instead, now that the CI is running in github itself.
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

Comment (dhoomakethu) [reply README.rst:5]: This gives the badge for `dev` branch
https://github.com/riptideio/pymodbus/actions/workflows/ci.yml/badge.svg?branch=dev

File: README.rst (hunk 1 above)
Comment (janiversen) [reply README.rst:35]: Done, in all places.

File: README.rst (hunk 2 above)
Comment (janiversen) [reply README.rst:64]: done in all places.

File: doc/TODO (hunk 4 above)
Comment (janiversen) [reply doc/TODO:27]: ups, corrected

File: README.rst (hunk 5 above)
Comment (janiversen) [reply README.rst:5]: Added.

File: README.rst (hunk 3 above)
Comment (janiversen) [reply README.rst]: Done.

**Comment by janiversen** [comment]
@dhoomakethu I should have all the changes you requested, I still have to see if I made the badge change correct.

Just out of curiosity, where are you based ?

File: README.rst (hunk 1 above)
Comment (dhoomakethu) [reply README.rst:36]: Thanks!!!

**Comment by dhoomakethu** [comment]
> @dhoomakethu I should have all the changes you requested, I still have to see if I made the badge change correct.
> 
> Just out of curiosity, where are you based ?

I am based out of India.

**Pull request merged**